week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,27 +1,118 @@
-function setAlarm() {}
+let timeLeft;
+let intervalId;
+let flashingIntervalId;
+let isPaused = false;
+let isRedBackground = false;
+let isBlueBackground = true;
 
-// DO NOT EDIT BELOW HERE
+function setAlarm() {
+  reset();
 
-function setup() {
-  var audio = new Audio("alarmsound.mp3");
+  timeLeft = Number(document.getElementById("alarmSet").value);
+  displayTimeRemaining(timeLeft);
+
+  intervalId = window.setInterval(intervalCallback, 1000);
+  function intervalCallback() {
+    if (!isPaused && timeLeft > 0 && timeLeft < 9999999999999999) {
+      timeLeft -= 1;
+    } else if (timeLeft === 0) {
+      playAlarm();
+      flashingIntervalId = window.setInterval(flashBackgroundColor, 500);
+      clearInterval(intervalId);
+    }
+    displayTimeRemaining(timeLeft);
+  }
+
+  if (timeLeft <= 0 || timeLeft > 999999999999999) {
+    alert("Please enter a number between 1 and 999999999999999");
+    reset();
+    document.getElementById("alarmSet").value = "";
+  }
+}
+
+function displayTimeRemaining(time) {
+  document.getElementById(
+    "timeRemaining"
+  ).innerText = `Time Remaining\n ${formatTime(time)}`;
+}
+
+function formatTime(numOfSecs) {
+  let hrsToDisplay = Math.floor(numOfSecs / 3600);
+  numOfSecs = numOfSecs % 3600;
+  let minsToDisplay = Math.floor(numOfSecs / 60);
+  let secsToDisplay = numOfSecs % 60;
+  return (
+    pad(hrsToDisplay) + ":" + pad(minsToDisplay) + ":" + pad(secsToDisplay)
+  );
+}
+
+function pad(num) {
+  return num < 10 ? "0" + num : num.toString();
+}
```

Comment:
I like how you split these three functions so that every one of them is not too big and does something specific 👌

BTW, note this exists in JS https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart